scripts/check_module_coverage_floor.py

```
@@ -0,0 +1,276 @@
+        if not match:
+            continue
+        path = match.group(1)
+        cover = float(match.group(6))
+        modules[path] = cover
+    return modules
+
+
+def load_baseline(path: Path) -> dict[str, Any]:
+    data = json.loads(path.read_text(encoding="utf-8"))
+    if "modules" not in data or not isinstance(data["modules"], dict):
+        raise ValueError(f"Invalid baseline file {path}: expected top-level 'modules' object")
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)